arcgis-ios-sdk-samples/Display information/Sketch on the map/SketchViewController.swift

```diff
@@ -16,78 +16,92 @@ import UIKit
+            self.redoBarButtonItem.isEnabled = self.sketchEditor.undoManager.canRedo
+            self.clearBarButtonItem.isEnabled = self.sketchEditor.geometry.map { !$0.isEmpty } ?? false
+        })
+        // Add the source code button item to the right of navigation bar.
         (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["SketchViewController"]
-        
-        self.sketchEditor = AGSSketchEditor()
-        self.mapView.sketchEditor = self.sketchEditor
-        
-        self.sketchEditor.start(with: nil, creationMode: .polyline)
-        
-        self.mapView.map = AGSMap(basemapStyle: .arcGISLightGrayBase)
-        self.mapView.interactionOptions.isMagnifierEnabled = true
-        
-        NotificationCenter.default.addObserver(self, selector: #selector(SketchViewController.respondToGeomChanged), name: .AGSSketchEditorGeometryDidChange, object: nil)
-        
-        // set viewpoint
-        self.mapView.setViewpoint(AGSViewpoint(targetExtent: AGSEnvelope(xMin: -10049589.670344, yMin: 3480099.843772, xMax: -10010071.251113, yMax: 3512023.489701, spatialReference: .webMercator())))
-    }
-    
-    @objc
-    func respondToGeomChanged() {
-        // Enable/disable UI elements appropriately
-        self.undoBBI.isEnabled = self.sketchEditor.undoManager.canUndo
-        self.redoBBI.isEnabled = self.sketchEditor.undoManager.canRedo
-        self.clearBBI.isEnabled = self.sketchEditor.geometry != nil && !self.sketchEditor.geometry!.isEmpty
     }
     
     // MARK: - Actions
     
-    @IBAction func geometryValueChanged(_ segmentedControl: UISegmentedControl) {
-        switch segmentedControl.selectedSegmentIndex {
-        case 0:// point
-            self.sketchEditor.start(with: nil, creationMode: .point)
-            
-        case 1:// polyline
-            self.sketchEditor.start(with: nil, creationMode: .polyline)
-            
-        case 2:// freehand polyline
-            self.sketchEditor.start(with: nil, creationMode: .freehandPolyline)
-            
-        case 3:// polygon
-            self.sketchEditor.start(with: nil, creationMode: .polygon)
-            
-        case 4:// freehand polygon
-            self.sketchEditor.start(with: nil, creationMode: .freehandPolygon)
-            
-        default:
-            break
+    @IBAction func addGeometryButtonTapped(_ sender: UIBarButtonItem) {
+        // Create an alert controller for the action sheets.
+        let alertController = UIAlertController(title: "Select a creation mode", message: nil, preferredStyle: .actionSheet)
+        // Create an action for each creation mode and add it to the alert controller.
+        creationModes.forEach { name, mode in
+            let action = UIAlertAction(title: name, style: .default) { (_) in
+                self.statusLabel.text = "\(name) selected."
+                self.sketchEditor.start(with: nil, creationMode: mode)
```

Comment:
This should be `[weak self]` in order to avoid the retain cycle.

Reply:
I agree `weak self` is better coding style. Here is a little story on why we used strong `self`... 😆 

We do sometimes capture a strong `self` when it is safe to do so, like in this case. Because `UIAlertAction` always has a shorter lifespan than the view controller itself, it would never/hardly incur a retain cycle. If to nitpick, we can put an `unowned self` here, but Phil mentioned a while ago that we prefer `weak self` over `unowned self`.

So in these unowned-able places we simply capture a strong `self` to leverage the new [Implicit capturing of self](https://github.com/apple/swift-evolution/blob/main/proposals/0269-implicit-self-explicit-capture.md#proposed-solution) syntax introduced in Swift 5.3. 😏 ~~and of course, the code is *significantly* shorter after applying this trick 🤣~~